xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
Perhaps to make it clearer:
T1:
1 Calls isrunning of t2
2 does sth with assumption t2 is there

T2:
1 join

If t2's join happens directly after t1's call to isrunning neither the
threadid nor the running status is worth anything without a more central
lock t1 and t2 share.
Am 04.07.2016 5:27 vorm. schrieb "Rainer Hochecker" <
notifications@github.com>:

> In xbmc/threads/Thread.cpp
> <https://github.com/xbmc/xbmc/pull/10063#discussion_r69405592>:
>
> > @@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
> >
> >  bool CThread::IsRunning() const
> >  {
> > +  CSingleLock lock(m_CriticalSection);
>
> making it atomic won't change behavior a bit.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/10063/files/57c6926939b3fc49e7538b4b6dc46bb19ae4d48d..327d4e188458dc489ea35197dbcafc53a6302bcc#r69405592>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCfHcCTzJO8cKcGepGffrVKoWwX87kjks5qSH2MgaJpZM4JDwSr>
> .
>